scopes/dependencies/modules/packages-excluder/generate-node-modules-pattern.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { generateNodeModulesPattern } from './generate-node-modules-pattern';
+
+describe('generateNodeModulesPattern()', () => {
+  it('should return an empty regex when the received array is empty', () => {
+    expect(generateNodeModulesPattern({ packages: [] })).toEqual('node_modules/(?!()/)');
+  });
+  it('should work with more than one packages', () => {
+    const packagesToTransform = ['react', '@myorg', 'testing-library__dom'];
+    const pattern = generateNodeModulesPattern({ packages: packagesToTransform });
+    const regex = new RegExp(pattern);
+    expect(pattern).toEqual(
+      'node_modules/(?!(react|.pnpm/registry.npmjs.org/react.*|.pnpm/react.*|@myorg|.pnpm/registry.npmjs.org/@myorg.*|.pnpm/@myorg.*|testing-library__dom|.pnpm/registry.npmjs.org/testing-library__dom.*|.pnpm/testing-library__dom.*)/)'
+    );
+    expect(regex.test('node_modules/react')).toBeTruthy();
+    expect(regex.test('node_modules/.pnpm/registry.npmjs.org/react')).toBeTruthy();
+    expect(regex.test('node_modules/.pnpm/react')).toBeTruthy();
+
+    expect(regex.test('node_modules/@myorg')).toBeTruthy();
+    expect(regex.test('node_modules/.pnpm/registry.npmjs.org/@myorg')).toBeTruthy();
+    expect(regex.test('node_modules/.pnpm/@myorg')).toBeTruthy();
+
+    expect(regex.test('node_modules/testing-library__dom')).toBeTruthy();
+    expect(regex.test('node_modules/.pnpm/registry.npmjs.org/testing-library__dom')).toBeTruthy();
+    expect(regex.test('node_modules/.pnpm/testing-library__dom')).toBeTruthy();
+  });
+  it('should work with one package', () => {
```

Comment:
when doing tests for one package and more than one package, always make the simple cases before the complex cases. so one package tests should come before multi packages.

Reply:
done

Reply:
done